### PR TITLE
Converts the string representation of an object of unsupported type i…

### PR DESCRIPTION
**Pull request by JohanMabille**
…nstead of throwing

### CODE REVIEW

File: test/test_pybind11_json.cpp [hunk 1]
```diff
@@ -145,6 +145,19 @@ namespace xpyt
...
     }
 
+    TEST(pyobject_tojson, class_)
+    {
+        py::scoped_interpreter guard;
+        py::object obj = py::globals();
+        nl::json j = obj;
+        ASSERT_TRUE(j.is_object());
+#if PY_MAJOR_VERSION==3
+        ASSERT_FALSE(j["__loader__"].get<std::string>().empty());
```
Comment (martinRenou) [inline test/test_pybind11_json.cpp]: I'm not sure I understand what you are testing here

File: include/pybind11_json/pybind11_json.hpp [hunk 2]
```diff
@@ -109,7 +109,7 @@ namespace nlohmann
                 }
                 return out;
             }
-            throw std::runtime_error("to_json not implemented for this type of object: " + obj.cast<std::string>());
+            return to_json_impl(py::repr(obj));
```
Comment (martinRenou) [inline include/pybind11_json/pybind11_json.hpp]: I am not really sure about this. I wonder if this is the same behavior as when using the Python `json` module. I remember it throwing when the serialization is not possible.

Comment (martinRenou) [reply include/pybind11_json/pybind11_json.hpp]: The Python `json` module throws an error for non-serializable objects:

`TypeError: Object of type module is not JSON serializable`

Comment (JohanMabille) [reply include/pybind11_json/pybind11_json.hpp]: ok let's keep this consistent with the Python module then.

**Pull request closed without merging**